src/db/tripledb.plt

```diff
@@ -1,54 +1,77 @@
+    assert_true(tripledb_ask(swrl_tests:'Adult', rdfs:'subClassOf', swrl_tests:'TestThing')).
 
 % delete individual triple
-test('tripledb_forget_from_triplestore(S,P,O)') :-
-  tripledb_forget(
-    knowrob:'Adult',
-    rdfs:'subClassOf',
-    knowrob:'TestThing').
+test('tripledb tests: forget triple') :-
+    assert_true(tripledb_forget(swrl_tests:'Adult', rdfs:'subClassOf', swrl_tests:'TestThing')).
 
 % check again for that triple and it should not be in the tripledb
-test('tripledb_ask_after_deletion', [fail]) :-
-  tripledb_ask(
-    knowrob:'Adult',
-    rdfs:'subClassOf',
-    knowrob:'TestThing').
-
-% add triple
-test('tripledb_tell_to_triplestore(S,P,O)') :-
-  tripledb_tell(
-    knowrob:'Adult',
-    rdfs:'subClassOf',
-    knowrob:'TestThing').
-
-
-% check again for that triple and it should be in the tripledb
-test('tripledb_ask_after_injection_of_new_triple') :-
-  tripledb_ask(
-    knowrob:'Adult',
-    rdfs:'subClassOf',
-    knowrob:'TestThing').
+test('tripledb tests: ask after deletion', [fail]) :-
+    tripledb_ask(swrl_tests:'Adult', rdfs:'subClassOf', swrl_tests:'TestThing').
+
+% add triple and check if it exists in db
+test('tripledb tests: tell to triplestore and check if triple exists') :-
+    assert_true(tripledb_tell(swrl_tests:'Adult', rdfs:'subClassOf', swrl_tests:'TestThing')),
+    assert_true(tripledb_ask(swrl_tests:'Adult', rdfs:'subClassOf', swrl_tests:'TestThing')).
+
+
+%%%%%%%%%%%%%%%%%%%%%%%% tests for xsd datatypes  %%%%%%%%%%%%%%%%%%%%%%%%
+% register namespace for following tests
+:- rdf_register_ns(test_datatype, 'http://knowrob.org/kb/datatype_test#',[keep(true)]).
+
+% load test owl file for tripledb testing for xsd datatypes
+test('tripledb tests: load local owl file for xsd datatypes') :-
+    tripledb_load(
+        'package://knowrob/owl/test/datatype_test.owl',
+        [ graph(test)
+        ]).
+
+% test for xsd:integer, Str, float
+test('tripledb tests: tell triples with various XSD DataTypes') :-
+    assert_true(tripledb_tell(test_datatype:'Lecturer3', test_datatype:'first_name', 'Johana$')),
+    assert_true(tripledb_tell(test_datatype:'Lecturer3', test_datatype:'last_name', 'Muller')),
+    assert_true(tripledb_tell(test_datatype:'Lecturer3', test_datatype:'studentId', 212123)),
+    assert_true(tripledb_tell(test_datatype:'Lecturer3', test_datatype:'height', 5.10)).
+
+% test for xsd:integer, Str, float
+test('tripledb tests: ask triples with various XSD DataTypes') :-
+    assert_true(forall(tripledb_ask(_, test_datatype:'studentId', X), integer(X))),
+    assert_true(forall(tripledb_ask(_, test_datatype:'first_name', Y), atom(Y))),
+    assert_true(forall(tripledb_ask(_, test_datatype:'last_name', Z), atom(Z))),
+    assert_true(forall(tripledb_ask(_, test_datatype:'height', H), float(H))).
+
+% remove the triple at the end of test
+test('tripledb tests: forget triple with various XSD DataTypes') :-
+    assert_true(tripledb_forget(test_datatype:'Lecturer3', _, _)).
```

Comment:
you should test if it was really removed. So do an ask before that succeeds, and an ask after that fails.